### DISCUSSION TITLE
Having issue with group-hover on background, bg color transitions not happening at the same time

### DISCUSSION THREAD

**Discussion by rcuttriss** [tailwindlabs/tailwindcss#10438] [Help] [answered]
HTML
`<button class="w-40 h-12 relative transform translate-x-1/2 text-2xl hover:bg-white group transition-colors">
            <span>Resumé</span> 
            <div class="pointer-lt-bg absolute block w-full h-full top-0 left-0 -z-10 group-hover:bg-white transition-colors"></div>
 </button>`

CSS
`.pointer-lt-bg {
	content: " ";
	transform-origin: bottom left;
	-ms-transform: skew(-30deg, 0deg);
	-webkit-transform: skew(-30deg, 0deg);
	transform: skew(-30deg, 0deg);
}`

I have been trying to have this slanted box have a simple white background on BOTH elements when I hover, I can accomplish this but the transitions for the elements do not seem to be lined up. Changing the duration of the transitions also doesn't seem to help.

**Comment by wongjn** [comment, marked as answer]
I think the background color transitions *are* happening at the same time and this is precisely causing the undesired behavior. The inner `<div>` element's background color is drawn on top of the outer `<button>`'s background color, and so when their background colors transition, the `<div>`'s background color is "added" on top of the `<button>`'s background color (since the color is transitioning from transparent) and so it *visually looks like* the transition completes sooner.

Some solutions to this I can think of is:
- Transition from a non-transparent color.
- Keep to one element, using `clip-path` to keep the same visual shape instead of an inner element.

[Here's a Tailwind Play of both solutions](https://play.tailwindcss.com/y2uho6pUzr).

**Reply by rcuttriss** [reply]
Thank you so much, and you even went as far as to code an example! I had no idea that the stacking of the transitions would give such an effect. I had not heard or used the `clip-path` yet and can already see the usefulness of it.
